gateway/ln-gateway/src/gateway_module_ng/send_sm.rs

```diff
@@ -0,0 +1,183 @@
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+pub enum SendSMState {
+    Sending(Sending),
+    Claiming([u8; 32], Vec<OutPoint>),
+    Cancelled,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Decodable, Encodable)]
+pub struct Sending {
+    pub max_delay: u64,
+    pub max_fee_msat: u64,
+    pub invoice: Bolt11Invoice,
+    pub claim_keypair: KeyPair,
+}
+
+impl State for SendStateMachine {
+    type ModuleContext = GatewayClientContextNG;
+
+    fn transitions(
+        &self,
+        context: &Self::ModuleContext,
+        global_context: &DynGlobalClientContext,
+    ) -> Vec<StateTransition<Self>> {
+        let gc = global_context.clone();
+
+        match &self.state {
+            SendSMState::Sending(sending) => {
+                vec![StateTransition::new(
+                    Self::send_payment(
+                        context.clone(),
+                        sending.max_delay,
+                        sending.max_fee_msat,
+                        sending.invoice.clone(),
+                    ),
+                    move |dbtx, result, old_state| {
+                        Box::pin(Self::transition_send_payment(
+                            dbtx,
+                            old_state,
+                            gc.clone(),
+                            result,
+                        ))
+                    },
+                )]
+            }
+            SendSMState::Claiming(..) => {
+                vec![]
+            }
+            SendSMState::Cancelled => {
+                vec![]
+            }
+        }
+    }
+
+    fn operation_id(&self) -> OperationId {
+        self.operation_id
+    }
+}
+
+impl SendStateMachine {
+    async fn send_payment(
+        context: GatewayClientContextNG,
+        max_delay: u64,
+        max_fee_msat: u64,
+        invoice: Bolt11Invoice,
+    ) -> Result<[u8; 32], ()> {
+        let lightning_context = loop {
+            match context.gateway.get_lightning_context().await {
+                Ok(lightning_context) => break lightning_context,
+                Err(_) => {
+                    sleep(std::time::Duration::from_secs(1)).await;
+                }
+            }
+        };
+
+        if lightning_context.lightning_public_key == invoice.recover_payee_pub_key() {
+            let invoice_msats = invoice
+                .amount_milli_satoshis()
+                .expect("We checked this previously");
+
+            return match context
+                .gateway
+                .receive(
+                    invoice.payment_hash().into_inner(),
+                    Amount::from_msats(invoice_msats),
+                )
+                .await
+            {
+                Ok(preimage) => Ok(preimage),
+                Err(e) => {
+                    error!("Failed to route internal payment {:?}", e);
+
+                    Err(())
+                }
+            };
+        }
+
+        match lightning_context
```

Comment:
In the legacy payment state machine, we check a bunch of parameters before actually making the payment in `await_get_payment_parameters` and `validate_outgoing_account`. We will want to add similar checks here.